src/components/Accordion/Accordion.tsx

```diff
@@ -0,0 +1,165 @@
+import AccordionItem from './AccordionItem'
+import type {AnimationType} from './ToggleAnimation'
+
+type ViewStyleProp = StyleProp<ViewStyle> | Array<StyleProp<ViewStyle>>
+type TextTitleStyleProp = StyleProp<TextStyle> | Array<StyleProp<TextStyle>>
+
+export type Section = {
+  [key: string]: string | number
+  title: string
+  content: string
+}
+
+export interface CommonAccordionProps {
+  /**
+   * onPress accordion event
+   */
+  onPress?: (keyExtractorItem: string) => void
+  /**
+   * render custom accordion header
+   */
+  renderHeader?: (
+    item: Section,
+    index: number,
+    expanded: boolean,
+    keyExtractorItem: string,
+  ) => React.ReactNode
+  /**
+   * render custom title section
+   */
+  renderSectionTitle?: (item: Section, index: number, expanded: boolean) => React.ReactNode
+  /**
+   * render custom content
+   */
+  renderContent?: (
+    item: Section,
+    index: number,
+    expanded: boolean,
+    keyExtractorItem: string,
+  ) => React.ReactNode
+  /**
+   * accordion title
+   */
+  title?: string
+  /**
+   * accordion container style
+   */
+  itemContainerStyle?: ViewStyleProp
+  /**
+   * accordion title style
+   */
+  titleStyle?: TextTitleStyleProp
+  /**
+   * accordion header style
+   */
+  headerContainerStyle?: ViewStyleProp
+  /**
+   * accordion body style
+   */
+  contentContainerStyle?: ViewStyleProp
+  /**
+   * animation type when open
+   * default: easeInEaseOut
+   */
+  openAnimation?: AnimationType
+  /**
+   * animation type when close
+   * default: easeInEaseOut
+   */
+  closeAnimation?: AnimationType
+  /**
+   * duration when open
+   * default: 300
+   */
+  openDuration?: number
+  /**
+   * duration when close
+   * default: 300
+   */
+  closeDuration?: number
+}
+
+export interface AccordionProps extends CommonAccordionProps {
+  /**
+   * section data sets
+   */
+  sections: Section[]
+  /**
+   * keyExtractor for Flatlist
+   */
+  keyExtractor?: (item: Section, index: number) => string
+  /**
+   * enable expand multiple item
+   * default: false
+   */
+  expandMultiple?: boolean
+  /**
+   * accordion wrapper style
+   */
+  wrapperStyle?: ViewStyleProp
+}
+
+const Accordion = React.forwardRef<FlatList, AccordionProps>(
+  ({sections, expandMultiple = false, keyExtractor, wrapperStyle, ...props}, ref) => {
+    const [array, setArray] = useState<string[]>([])
+
+    const _keyExtractor = useMemo(
+      () => keyExtractor || ((_: Section, index: number) => index.toString()),
+      [keyExtractor],
+    )
+
+    const onPress = useCallback(
+      (key: string) => {
+        const index = array.findIndex(e => e === key)
+        let newArray = [...array]
+
+        if (expandMultiple) {
+          if (index >= 0) {
+            newArray.splice(index, 1)
+          } else {
+            newArray.push(key)
+          }
+        } else {
+          newArray = index >= 0 ? [] : [key]
+        }
+
+        setArray(newArray)
+      },
+      [array, expandMultiple],
```

Comment:
I think we shouldn't add `array` to the dependency because on line 128 you updated the `array` so this function will be re-render.
we should you this approach to prevent this
`setArray(oldArray => {
  const newArray = [] //add your logic
  return newArray
})
`

Reply:
updated